src/ucs/memory/memtype_cache.h

```diff
@@ -22,10 +23,18 @@ typedef struct ucs_memtype_cache         ucs_memtype_cache_t;
 typedef struct ucs_memtype_cache_region  ucs_memtype_cache_region_t;
 
 
+/* Memory information record */
+typedef struct ucs_memory_info {
+    uint8_t          type;    /**< Memory type, use uint8 for compact size */
+    ucs_sys_device_t sys_dev; /**< System device index */
```

Comment:
As mentioned earlier, does it make sense to add base_addr, alloc_length, buffer_id fields into this struct in a follow up PR?

Reply:
same as https://github.com/openucx/ucx/pull/6180#discussion_r567247764